caffe2/serialize/inline_container.h

```diff
@@ -119,135 +134,202 @@ class PyTorchStreamReader {
     auto ptr = malloc(size);
     at::DataPtr retval(ptr, ptr, free, at::kCPU);
 
-    in.read((char*)ptr, size);
-    cursor += size;
+    in_.read((char*)ptr, size);
+    cursor_ += size;
     seekToNextAlignmentBoundary();
     return std::tuple<at::DataPtr, size_t>(std::move(retval), size);
   }
+
+  // return dataptr, key, size
+  std::tuple<at::DataPtr, int64_t, int64_t> getNextRecord() {
+    int64_t key = cursor_;
+    if (!hasNextRecord()) {
+      throw std::runtime_error("No more record, but hasNextRecord is called.");
+    }
+    auto tag = read64BitIntegerLittleEndian();
+    if (tag != RecordTags::STORAGE) {
+      throw std::runtime_error(
+          "Attempted to read a record of non-storage type");
+    }
+    auto size = read64BitIntegerLittleEndian();
+    seekToNextAlignmentBoundary();
+    auto ptr = malloc(size);
```

Comment:
The return value type of malloc is void*. It's not a good idea to use auto here. At least to use auto* here to indicate it's a pointer.